c/authService.c

```diff
@@ -173,16 +188,209 @@ static int serveAuthCheck(HttpService *service, HttpResponse *res) {
+  } else {
+    zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_DEBUG2,
+           "RegEx compilation error %s.", regexStr);
+  }
+  value = regexec(&regex, url, 0, NULL, 0);
+  char urlCpy[1024];
+  snprintf(urlCpy, 1024, url);
+  int index = 0;
+  while (urlCpy[index]) { // Capitalize query
+      urlCpy[index] = toupper(urlCpy[index]);
+      index++;
+  }
+  if (instanceID < 0) { // Set instanceID
+    instanceID = 0;
+  }
+  if (value == REG_NOMATCH) {
+    zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_DEBUG2,
+           "RegEx didn't match.");
+    char * token = strtok(urlCpy, "/");
+    int subUrlIndex = -1;
+    while( token != NULL ) {
+      if (strcmp(rootServiceName, NULL) == 0)
+      {
+        snprintf(rootServiceName, 1024, token);
+      } else {
+        snprintf(subUrl[subUrlIndex], 1024, token);
+      }
+      subUrlIndex++;
+      token = strtok(NULL, "/");
+    }
+    snprintf(productCode, 1024, "ZLUX");
+    snprintf(type, 1024, "core");
+  }
+  else if (!value) {
+    char * token = strtok(urlCpy, "/");
+    int subUrlIndex;
+    subUrlIndex = 0;
+    while( token != NULL ) {
+      switch(subUrlIndex) {
+        case 0:
+          snprintf(productCode, 1024, token);
+          break;
+        case 1:
+          snprintf(_p, 1024, token);
+          break;
+        case 2:
+          snprintf(pluginID, 1024, token);
+          break;
+        case 3:
+          snprintf(_s, 1024, token);
+          break;
+        case 4:
+          snprintf(serviceName, 1024, token);
+          break;
+        case 5:
+          snprintf(_v, 1024, token);
+          break;
+        default:
+          snprintf(subUrl[subUrlIndex-6], 1024, token); // subtract 6 from maximum index to begin init subUrl array at 0
+      }
+      
+      subUrlIndex++;
+      token = strtok(NULL, "/");
+    }
+    if ((strcmp(pluginID, "ORG.ZOWE.CONFIGJS") == 0) && (strcmp(serviceName, "DATA") == 0))
+    {
+      snprintf(type, 1024, "config");
+      snprintf(pluginID, 1024, subUrl[0]);
+      snprintf(scope, 1024, subUrl[1]);
+      
+    } else {
+      snprintf(type, 1024, "service");
+    }
+    char* ch; 
+    char* chReplace;
+    ch = ".";
+    chReplace = "_";
+    for (index = 0; index <= strlen(pluginID); index++)
+  	{
+  		if (pluginID[index] == *ch)  
+      {
+        pluginID[index] = *chReplace;
+      }
+    }
+    zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_DEBUG2,
+           "RegEx match OK.");
+  }
+  else {
+    zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_WARNING,
+           "RegEx match failed.");
+  }
+  snprintf(profileName, 1024, makeProfileName(type, 
+    productCode, 
+    instanceID, 
+    pluginID, 
+    rootServiceName,
+    serviceName,
+    method,
+    scope,
+    subUrl));
+  // printf("\n\nFinal query profileName & URL %s - %s\n\n", profileName, url);
+  
+  /* Free memory allocated to the pattern buffer by regcomp() */
+  regfree(&regex);
+  
+  return profileName;
+}
+
+const char* makeProfileName(
```

Comment:
Is this function going to be used outside this source file? If not, make it static.